### ISSUE TITLE
False-positive `import-error` for platform specific modules in platform check

### ISSUE THREAD

**Issue by nineteendo** [pylint-dev/pylint#9610]
### Bug description

```python
# pylint: disable=missing-module-docstring
import sys

if sys.platform == "win32":
    import msvcrt
    print(msvcrt.kbhit())
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:5:4: E0401: Unable to import 'msvcrt' (import-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.1.0.2.5)]
```

### OS / Environment

macOS 13.6.6 (22G630)

**Comment by Pierre-Sassoulas** [comment]
Isn't this is a duplicate of https://github.com/pylint-dev/pylint/issues/7240 ?

**Comment by nineteendo** [comment]
At least it's related to platform checking. If you want, you can track it further there.
